### PR TITLE
90381 facility search

### PR DESCRIPTION
**Pull request by coope93**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to Summary and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- UI enhancements for the 1010CG Facility Search. We are not utilizing the va-search-input component. This required converting the page to a custom page.
- Added ability to see 5 more facilities at a time. 
- The facilities selection does not currently render on the confirmation page as a result of using the new custom page. We will have a subsequent ticket to address this. 
- The changes are behind a flipper toggle [`caregiver_use_facilities_API`]

## Related issue(s)
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/90381

## Testing done

- Uses the va-search-input component to search and load facilities. 

## Screenshots

| Before | After |
| ------- | ------ |
| Screenshot 2024-09-03 at 2 18 15 PM|Screenshot 2024-09-03 at 2 23 29 PMScreenshot 2024-09-04 at 10 21 13 AM

## What areas of the site does it impact?

10-10CG

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/caregivers/actions/fetchFacilities.js [hunk 1]
```diff
@@ -46,15 +46,15 @@ export const fetchFacilities = async ({
...
+    .catch(error => {
       Sentry.withScope(scope => {
         scope.setExtra('error', error);
         Sentry.captureMessage(content['error--facilities-fetch']);
       });
 
       return {
         type: 'SEARCH_FAILED',
-        errorMessage: error,
+        errorMessage: 'There was an error fetching the health care facilities.',
```
Comment (coope93) [inline src/applications/caregivers/actions/fetchFacilities.js:57]: We still need confirmation on what the wording for this generic error message should be when the facilities request returns an error.

File: src/applications/caregivers/components/FormFields/FacilityList.jsx [hunk 2]
```diff
@@ -48,13 +53,6 @@ const FacilityList = props => {
     />
   ));
 
-  useEffect(
```
Comment (coope93) [inline src/applications/caregivers/components/FormFields/FacilityList.jsx:51]: This was causing the page to jump to the top of the list when the `load more facilities` was loading more facilities. Now we let the parent component handle when to focus on the results.

**Review by mkerns1 — DISMISSED** [review]
LGTM!

File: src/applications/caregivers/components/FormFields/FacilitySearch.jsx [hunk 3]
```diff
@@ -37,119 +62,212 @@ const FacilitySearch = props => {
...
-        <va-loading-indicator
-          label={content['app-loading-generic-text']}
-          message={content['facilities-loading-text']}
-          set-focus
-        />
+        <>
+          <FacilityList {...facilityListProps} />
+          {loadingMoreFacilities && loader()}
+          {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+          <a href="#" onClick={showMoreFacilities}>
```
Comment (longmd) [inline src/applications/caregivers/components/FormFields/FacilitySearch.jsx:193]: We need to refactor this to a `button`. This is an a11y violation. I assume you want it to look like a link? If so, we have a class for that. Lets do this as:
```
<button type="button" className="va-button-link" onClick={showMoreFacilities}>
    Load more facilities
</button>
```

Comment (longmd) [inline src/applications/caregivers/components/FormFields/FacilitySearch.jsx:193]: We need to refactor this to a `button`--as this is an a11y violation. I assume you want it to look like a link? If so, we have a class for that. Lets do this as:
```
<button type="button" className="va-button-link" onClick={showMoreFacilities}>
    Load more facilities
</button>
```

File: src/applications/caregivers/config/form.js [hunk 4]
```diff
@@ -55,6 +54,7 @@ import FacilityConfirmation from '../components/FormPages/FacilityConfirmation';
 // sign as representative
 import signAsRepresentativeYesNo from './chapters/signAsRepresentative/signAsRepresentativeYesNo';
 import uploadPOADocument from './chapters/signAsRepresentative/uploadPOADocument';
+import FacilitySearch from '../components/FormFields/FacilitySearch';
```
Comment (longmd) [inline src/applications/caregivers/config/form.js:57]: Can we nest this with the appropriate "chapter" pages--so up around L29-30?

**Review by longmd — CHANGES_REQUESTED** [review]

**Review by longmd — APPROVED** [review]
LGTM 🚀

**Pull request merged**